### PR TITLE
feat: adds-host-cli-option

### PR DESCRIPTION
**Pull request by dutterbutter**
# What :computer: 
* `src/config/cli.rs`: Added a new `host` field to the `Cli` struct, allowing the server to listen on multiple IP addresses. 
* Updated the `impl Cli` to include the new `host` field in the configuration.
* `src/config/mod.rs`: Added a new `host` field to the `TestNodeConfig` struct and updated its default implementation. Modified the `print` method to display all hosts and added a new `with_host` method to set the hosts.
* `src/main.rs`: Updated the `main` function to handle multiple hosts by iterating over the `config.host` field and creating a `SocketAddr` for each host.

# Why :hand:
* Aligns with Anvil
* Adds flexibility in configurability 

# Evidence :camera:
Include screenshots, screen recordings, or `console` output here demonstrating that your changes work as intended

### CODE REVIEW

File: src/config/cli.rs [hunk 1]
```diff
@@ -39,6 +40,16 @@ pub struct Cli {
...
 
+    /// The hosts the server will listen on.
+    #[arg(
+        long,
+        value_name = "IP_ADDR",
+        default_value = "127.0.0.1",
+        value_delimiter = ',',
+        help_heading = "Network Options"
+    )]
+    pub host: Vec<IpAddr>,
```
Comment (itegulov) [inline src/config/cli.rs:51]: Comparing this to upstream anvil I have noticed that we don't match env vars they use. Is there a reason for that?

**Review by itegulov — APPROVED** [review]
LGTM, just one small nit

File: src/config/cli.rs (hunk 1 above)
Comment (dutterbutter) [reply src/config/cli.rs:51]: Not intentionally left out. Will add it so theres full parity. 🙏

Comment (dutterbutter) [reply src/config/cli.rs:58]: Updated ✅

**Review by itegulov — APPROVED** [review]

**Pull request merged**